inst/include/sampler.h

```diff
@@ -273,10 +274,21 @@ namespace smc {
         Moves = sFrom.Moves;
         /// The additional algorithm parameters.
         algParams = sFrom.algParams;
-        /// An object for adapting additional algorithm parameters
-        pAdapt = sFrom.pAdapt;
-        ///A flag to track whether the adaptation object needs to be included in this destructor.
-        adaptBelong = sFrom.adaptBelong;
+        if(sFrom.adaptBelong) {
+            // this can only happen if the default adaptMethods was used,
+            // i.e., no call to SetAdaptMethods
+            pAdapt = new adaptMethods<Space,Params>;
+            adaptBelong = 1;
```

Comment:
Don't we also need a
`*pAdapt = *sFrom.pAdapt`
here?

I'd expect a copy of a sampler to end up in exactly the same configuration as the one that was being copied.

Reply:
I would have thought that this one would be okay. If adaptBelong is 1 then SetAdaptMethods hasn't been called at any point so pAdapt is empty. I *think* the above will produce something that's equivalent but independent.

Reply:
That might be true with the current implementation, but there's a certain lack of robustness in assuming you can trace the execution path to the current configuration. Any future changes could break things in a way that would be rather subtle.

In particular, I think it's going to be difficult to address the other issue without introducing a mechanism by which a copied sampler can end up owning pAdapt when the object being copied did not.

As `adaptBelong` only indicates memory ownership I'd be inclined to include an explicit copy even if it may be redundant at this instant.